server/controllers/authController.js

```diff
@@ -181,45 +183,94 @@ This link will expire in 24 hours.`)
       .catch(err => res.status(500).json({message: errorMessage(err)}));
   };
 
-  this.loginUser = async function (req, res) {
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-      res.status(400).json({
-        message: "Invalid request",
-        errors: errors.array()
-      });
-      return;
-    }
-    // Check the credentials
-    let user;
-    try {
-      user = await userRepository.getByEmail(req.body.email);
-    } catch (err) {
-      res.status(500).json({message: errorMessage(err)});
-      return;
-    }
-    if (!user || !validatePassword(req.body.password, user.password)) {
-      res.status(400).json({message: "Invalid username/password"});
-      return;
-    }
-    const lastLogin = user.lastLogin;
-    const currentDate = moment();
-    await userRepository.update(user, {
-      lastLogin: currentDate
-    })
-      .then(() => {
-        res.status(200).json({
-          message: "Successful login!",
-          user: {
-            uid: user.id,
-            isAdmin: user.isAdmin,
-            lastLogin: lastLogin,
-            token: generateToken(user)
-          }
-        });
+    this.loginUser = async function (req, res) {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
+        res.status(400).json({message: "Invalid request", errors: errors.array()});
+        return;
+      }
+      // Check there is a user
+      let user;
+      try {
+        user = await userRepository.getByEmail(req.body.email);
+      } catch (err) {
+        res.status(500).json({message: errorMessage(err)});
+        return;
+      }
+      if (!user) {
+        res.status(400).json({message: "Invalid Username/Password. Please try again."});
+        return;
+      }
+      // Check if account is locked
+      const currentDate = moment();
+      if (user.unlockDate && currentDate.isBefore(user.unlockDate)) {
+        res.status(400).json({message: "Invalid Username/Password. Please try again."});
```

Comment:
We're trying to avoid stating when an account is locked on the front-end just because otherwise its a way for an attacker to identify that an email is a valid account

Reply:
If the account is locked, then they should receive an email stating so. But it is basically trying to establish the balance between convenience and security.

Reply:
Do you think we should just say account is locked if we receive any login requests for locked accounts.